src/main/java/com/gym/gymsystem/config/SecurityConfig.java

```diff
@@ -0,0 +1,89 @@
+package com.gym.gymsystem.config;
+
+import com.gym.gymsystem.filter.JwtAuthenticationFilter;
+import com.gym.gymsystem.security.CustomAuthenticationEntryPoint;
+import com.gym.gymsystem.security.CustomAuthenticationProvider;
+import com.gym.gymsystem.service.CustomUserDetailsService;
+import com.gym.gymsystem.service.TokenBlacklistService;
+import com.gym.gymsystem.util.JwtTokenUtil;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+    private final CustomAuthenticationEntryPoint entryPoint;
+    private final CustomAuthenticationProvider authenticationProvider;
+    private final JwtTokenUtil jwtTokenUtil;
+    private final CustomUserDetailsService customUserDetailsService;
+    private final TokenBlacklistService tokenBlacklistService;
+
+    public SecurityConfig(CustomAuthenticationEntryPoint entryPoint,
+                          @Lazy CustomAuthenticationProvider authenticationProvider, JwtTokenUtil jwtTokenUtil, CustomUserDetailsService customUserDetailsService, TokenBlacklistService tokenBlacklistService) {
+        this.entryPoint = entryPoint;
+        this.authenticationProvider = authenticationProvider;
+        this.jwtTokenUtil = jwtTokenUtil;
+        this.customUserDetailsService = customUserDetailsService;
+        this.tokenBlacklistService = tokenBlacklistService;
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers(HttpMethod.POST, "/api/trainee").permitAll()
+                        .requestMatchers(HttpMethod.POST, "/api/trainer").permitAll()
+                        .requestMatchers( "/api/auth/change-password").permitAll()
```

Comment:
`/api/ `is repeatable in your endpoints, you can use '`server.servlet.context-path`' for this